nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -618,17 +618,28 @@ def collate_fn(self, batch):
 
         position_ids: List[List[int]] = []
         cu_seqlens: List[List[int]] = []
+        cu_seqlens_unpadded: List[List[int]] = []
         for item in batch:
             position_ids.append([])
             cu_seqlens.append([0])
+            cu_seqlens_unpadded.append([0])
             seqlens = np.array(item['seq_boundaries'][1:]) - np.array(item['seq_boundaries'][:-1])
             for l in seqlens:
                 # length minus 1 because input_ids is truncated by 1 for labels
                 position_ids[-1].extend(list(range(l - 1)))
                 cu_seqlens[-1].append(cu_seqlens[-1][-1] + l - 1)
-            # set last seq to the max seq len because rope and attn kernels expect no padding
+
+            # the last seq needs to be the max seq len because rope and attn kernels expect no padding
             cu_seqlens[-1][-1] = max_length
```

Comment:
Noticed that you have `cu_seqlens_unpadded`, so your `cu_seqlens` is actually `cu_seqlen_padded`? If so, I have same question as before, why overwrite the last element directly? shouldn't you append max_length instread?

I see you did this in [this commit](https://github.com/NVIDIA/NeMo/pull/10688/commits/850d9ae5ac485d8a1049981e67ccd008e5622a29), why did you recover the change later?

Reply:
Not sure why it was left out. I have reverted this part back to what I had in the [this commit](https://github.com/NVIDIA/NeMo/pull/10688/commits/850d9ae5ac485d8a1049981e67ccd008e5622a29)